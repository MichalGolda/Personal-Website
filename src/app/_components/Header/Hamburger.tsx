export type HamburgerProps = {
  isShown: boolean;
  onClick: () => void;
};

export function Hamburger({ isShown, onClick }: HamburgerProps) {
  return (
    <div
      onClick={onClick}
      className="flex flex-col gap-y-2 cursor-pointer lg:hidden"
    >
      <div
        className={`${
          isShown && "rotate-45 translate-y-[4px] translate-x-[-4px]"
        } block w-8 bg-secondary h-0.5 transition-transform duration-500`}
      ></div>
      <div
        className={`${
          isShown && "-rotate-45 translate-y-[-5px] translate-x-[-4px]"
        } block w-8 bg-secondary h-0.5 transition-transform duration-500`}
      ></div>
    </div>
  );
}
