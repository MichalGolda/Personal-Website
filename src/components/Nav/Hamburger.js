import {
  StyledHamburgerContainer,
  StyledHamburgerSpan,
} from "./Hamburger.styled";

export default function Hamburger() {
  return (
    <StyledHamburgerContainer>
      <StyledHamburgerSpan />
      <StyledHamburgerSpan />
      <StyledHamburgerSpan />
    </StyledHamburgerContainer>
  );
}
