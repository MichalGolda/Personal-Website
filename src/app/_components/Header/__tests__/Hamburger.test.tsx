import "@testing-library/jest-dom";
import { render, fireEvent } from "@testing-library/react";
import { Hamburger } from "../Hamburger";

describe("Hamburger component", () => {
  it("renders without errors", () => {
    const onClickMock = jest.fn();

    const { container } = render(
      <Hamburger isShown={true} onClick={onClickMock} />
    );

    const hamburger = container.firstChild;

    expect(hamburger).toBeInTheDocument();
    expect(hamburger).toHaveClass("lg:hidden");

    // @ts-ignore
    fireEvent.click(hamburger);

    expect(onClickMock).toHaveBeenCalled();
  });

  it("renders unchanged", () => {
    const { container } = render(
      <Hamburger isShown={true} onClick={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });
});
