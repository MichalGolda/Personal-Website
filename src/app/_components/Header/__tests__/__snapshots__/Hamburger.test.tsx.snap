// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger component renders unchanged 1`] = `
<div>
  <div
    class="flex flex-col gap-y-2 cursor-pointer lg:hidden"
  >
    <div
      class="rotate-45 translate-y-[4px] translate-x-[-4px] block w-8 bg-secondary h-0.5 transition-transform duration-500"
    />
    <div
      class="-rotate-45 translate-y-[-5px] translate-x-[-4px] block w-8 bg-secondary h-0.5 transition-transform duration-500"
    />
  </div>
</div>
`;
